src/interfaces/index.ts

```diff
@@ -0,0 +1,83 @@
+import { USER_SORT, USER_FILTER, USER_FETCH, INPUT_SEARCH, USER_FETCH_REJECTED, USER_FETCH_FULFILLED } from '../type';
+import { ChangeEvent } from 'react';
+export interface IAction {
+    type? : string;
+    payload? : string & Array<IUser> & IError;
```

Comment:
To make this consistent across all actions, the payload should always be an object. Not string or number.